client/layout/logged-out.jsx

```diff
@@ -162,7 +163,7 @@ const LayoutLoggedOut = ( {
 
 	if ( useOAuth2Layout && ( isGravatar || isGravPoweredClient ) ) {
 		masterbar = null;
-	} else if ( useOAuth2Layout && oauth2Client && oauth2Client.name ) {
+	} else if ( useOAuth2Layout && oauth2Client && oauth2Client.name && ! masterbarIsHidden ) {
```

Comment:
Big win if we can effectively remove this logic. 
Created an issue for it: https://linear.app/a8c/issue/DS-230/design-system-calypso-enable-and-unify-the-master-bar-and-wp-logo. I feel that is indeed the intention with the DS - to have a single/unified master bar as well 🤔